modules/imgproc/test/test_color.cpp

```diff
@@ -3072,20 +3072,43 @@ TEST(ImgProc_RGB2YUV, regression_13668)
     EXPECT_EQ(res, ref);
 }
 
-TEST(ImgProc_cvtColorTwoPlane, missing_check_17036)  // test can be removed if required feature is implemented
+TEST(ImgProc_cvtColorTwoPlane, regression_17036)
 {
-    std::vector<uchar> y_data(700 * 480);
-    std::vector<uchar> uv_data(640 * 240);
+    RNG &rng = theRNG();
 
-    Mat y_plane_padding(480, 640, CV_8UC1, y_data.data(), 700);  // with stride
-    Mat uv_plane(240, 320, CV_8UC2, uv_data.data());
+    std::vector<uchar> y_reference(640 * 480);
+    std::vector<uchar> uv_reference(640 * 240);
+    std::vector<uchar> y_padded(700 * 480);
+    std::vector<uchar> uv_padded(700 * 240);
 
-    Mat result;
+    Mat y_reference_mat(480, 640, CV_8UC1, y_reference.data());
+    Mat uv_reference_mat(240, 320, CV_8UC2, uv_reference.data());
+    Mat y_padded_mat(480, 640, CV_8UC1, y_padded.data(), 700);
+    Mat uv_padded_mat(240, 320, CV_8UC2, uv_padded.data(), 700);
+
+    rng.fill(y_reference_mat, RNG::UNIFORM, 16, 235);
+    rng.fill(uv_reference_mat, RNG::UNIFORM, 16, 240);
+
+    for (int i = 0; i < y_reference_mat.rows; ++i) {
+        for (int j = 0; j < y_reference_mat.cols; j++) {
+            y_padded_mat.at<uchar>(i, j) = y_reference_mat.at<uchar>(i, j);
+        }
+    }
```

Comment:
`y_reference_mat.copyTo(y_padded_mat(Rect(0, 0, y_reference_mat.cols, y_reference_mat.rows)));`

Reply:
Thanks :+1:

Reply:
When using copyTo we face the following error:
`unknown file: Failure
C++ exception with description "OpenCV(3.4.13-dev) /build/precommit_linux64/3.4/opencv/modules/core/src/matrix.cpp:751: error: (-215:Assertion failed) 0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows in function 'Mat'
" thrown in the test body.`